src/olympia/api/authentication.py

```diff
@@ -20,16 +21,23 @@ class JSONWebTokenAuthentication(UpstreamJSONWebTokenAuthentication):
     """
     DRF authentication class for JWT header auth.
     """
-    def authenticate_credentials(self, request):
+
+    def authenticate_credentials(self, payload):
         """
         Mimic what our ACLMiddleware does after a successful authentication,
         because otherwise that behaviour would be missing in the API since API
         auth happens after the middleware process request phase.
         """
-        result = super(
-            JSONWebTokenAuthentication, self).authenticate_credentials(request)
-        amo.set_user(result)
-        return result
+        try:
+            user = UserProfile.objects.get(pk=payload['user_id'])
+        except UserProfile.DoesNotExist:
+            raise exceptions.AuthenticationFailed('User not found.')
+
+        if user.deleted:
```

Comment:
We don't seem to use `user.is_active`. For some reason I can't even update its value in the database so I can't test this. We don't check it anywhere else so I don't think we need to check it here. We "delete" users with `deleted` not `is_active`.